### PR TITLE
[feature] Add command line option to enable goQuery profiling

### LINKED ISSUE #168: Add profiling options to goQuery
I just realized that goQuery does not have a profiling parameter / option (in the past I've always added it manually to the code whenever I wanted to make assessments). In the light of supporting PGO (and also for debugging / performance measurements) I think it would be prudent to add an option similar to what we had for goProbe before it was replaced with the API level profiling (as in: one parameter for a profile directory, triggering both CPU and memory profile output for the goQuery invocation).

### PR DESCRIPTION
**Pull request by fako1024**
Simple enough, basically re-uses proven code recently removed from goProbe after moving to API based profiling.

Closes #168

### CODE REVIEW

File: cmd/goQuery/cmd/root.go [hunk 1]
```diff
@@ -90,6 +90,8 @@ func init() {
 	flags.BoolVarP(&cmdLineParams.Sum, "sum", "", false, helpMap["Sum"])
 	flags.BoolVarP(&cmdLineParams.Version, "version", "v", false, "Print version information and exit\n")
 
+	flags.StringVar(&cmdLineParams.ProfilingOutputDir, "profiling-output-dir", "", "Enable and set directory to store CPU and memory profiles")
```
Comment (els0r) [inline cmd/goQuery/cmd/root.go:93]: I wouldn't make this a property of the query, but rather list this in pkg/conf.

E.g.: const ProfilingOutputDir = profilingKey + ".output-dir"

**Review by els0r — CHANGES_REQUESTED** [review]
Profiling logic looks sound. Would appreciate the change for the flags

File: cmd/goQuery/cmd/root.go (hunk 1 above)
Comment (fako1024) [reply cmd/goQuery/cmd/root.go]: Sure think. Like this: 235c67c2b166ee0a085616ee6fa17589cd94b2a5 ?

Comment (els0r) [reply cmd/goQuery/cmd/root.go]: Almost 😄 . Let me make a suggestion before we ping-pong.

Comment (els0r) [reply cmd/goQuery/cmd/root.go]: @fako1024 : please see latest push regarding the re-arrangement of the function calls.

File: cmd/goQuery/cmd/root.go [hunk 2]
```diff
@@ -32,13 +32,16 @@ var cfgFile string
...
-	Long:          helpBaseLong,
-	RunE:          entrypoint,
-	Args:          validatePositionalArgs,
-	SilenceUsage:  true,
-	SilenceErrors: true,
+	Use:   "goQuery -i <interfaces> QUERY TYPE",
+	Short: helpBase,
+	Long:  helpBaseLong,
+	// we want to make sure that every command can be profiled
+	PersistentPreRunE:  initProfiling,
```
Comment (els0r) [inline cmd/goQuery/cmd/root.go:39]: The use of `PersistentPreRun` will make sure that this set up code is called before _every_ command, not just the query commands.

This is relevant so we also get profiles if `list` is called.

**Review by els0r — APPROVED** [review]
Feel free to merge if you agree.

File: cmd/goQuery/cmd/root.go (hunk 2 above)
Comment (fako1024) [reply cmd/goQuery/cmd/root.go:39]: I see, thanks for cross-checking / proposing the change! Good to go then?

**Pull request merged**